### PR TITLE
[grapl-web-ui] Add integration testing for authenticated web APIs

### PR DESCRIPTION
**Pull request by inickles-grapl**
### What changes does this PR make to Grapl? Why?

Primarily this adds the ability to test authenticated APIs in the grapl-web-ui. Each integration test can initialize a new instance of TestApp, which provisions a test user in the auth database for the test. This adds an integration test for the check_login API when the user is authenticated. 

Additionally, this PR has a number of other small changes:
- Increase memory of Nomad jobs for web-ui and rust-intergation-tests - was getting OOM errors.
- I removed what looks to be similar authentication tests from Python.
- Add a health check to grapl-web-ui at `api/health' and configure Nomad to use that to determine service health.
- Add a ConfigBuilder to make it easier to configure the bind address.

### How were these changes tested?

I ran the Rust integration tests and verified they were running.

### CODE REVIEW

File: src/rust/Dockerfile [hunk 1]
```diff
@@ -157,7 +157,7 @@ RUN --mount=type=cache,target="${CARGO_TARGET_DIR}",sharing=locked \
         --features "${RUST_INTEGRATION_TEST_FEATURES}" \
         --no-run \
         --message-format=json \
-        --test "*" | \
+        --test "*" -- --nocapture | \
```
Comment (inickles-grapl) [inline src/rust/Dockerfile:160]: do not capture stdout from tests

File: nomad/rust-integration-tests.nomad [hunk 2]
```diff
@@ -159,6 +154,11 @@ job "rust-integration-tests" {
               destination_name = "event-source"
               local_bind_port  = 1005
             }
+
+            upstreams {
+              destination_name = "web-ui"
+              local_bind_port  = 1007
```
Comment (wimax-grapl) [inline nomad/rust-integration-tests.nomad:160]: noti mportant, but why not 1006

Comment (inickles-grapl) [reply nomad/rust-integration-tests.nomad:160]: Ah, no good reason: originally I had another upstream using 1006, I just hadn't updated this one after removing the other. I'll fix that to use 1006.

File: src/rust/grapl-web-ui/tests/api/auth/check_login.rs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#[actix_web::test]
+async fn auth_unauthenticated_check_login() -> eyre::Result<()> {
+    let app = crate::test_app::TestApp::init().await?;
+
+    let response = app.post("api/auth/checkLogin").send().await?;
+
+    println!("{:?}", response);
```
Comment (wimax-grapl) [inline src/rust/grapl-web-ui/tests/api/auth/check_login.rs:7]: don't love the bare println but whatever

File: src/rust/grapl-web-ui/tests/api/config.rs [hunk 4]
```diff
@@ -0,0 +1,56 @@
+use grapl_config::env_helpers::FromEnv;
+use rusoto_dynamodb::DynamoDbClient;
+
+type Result<T> = std::result::Result<T, TestConfigError>;
+
+pub struct TestConfig {
```
Comment (wimax-grapl) [inline src/rust/grapl-web-ui/tests/api/config.rs:6]: fwiw, in the rest of our codebase we've mostly been using Clap for inputting env var stuff like this, but no biggie either way

**Review by wimax-grapl — APPROVED** [review]

File: src/rust/grapl-web-ui/tests/api/config.rs (hunk 4 above)
Comment (inickles-grapl) [reply src/rust/grapl-web-ui/tests/api/config.rs:6]: Yeah, that's a good point - it's what I should have done. I think I'll make that a separate PR after this, where I'll migrate this and the main app to clap as well.

I migrated the main app to use clap in a later draft PR (https://github.com/grapl-security/grapl/pull/1839) - I'll start by bringing that part into its own.

File: src/rust/grapl-web-ui/tests/api/auth/check_login.rs (hunk 3 above)
Comment (inickles-grapl) [reply src/rust/grapl-web-ui/tests/api/auth/check_login.rs:7]: Yeah, they could at least say what they are. 👍

**Pull request merged**